### PR TITLE
Feature/linting

### PR DESCRIPTION
**Pull request by NegativeK**
Select all lint rules with a handful of specific exceptions.

Clean up lint and typing errors. Implement a simple Makefile for static checking.

Issue #14.

### CODE REVIEW

**Comment by mrjones-plip** [comment]
Are you assigning this ticket to me or requesting a review?  I think the latter, yes? 

In my mind:

Assignee: person doing the coding
Reviewer: person reviewing the code (optionally adding some commits if Assignee is cool with that)

**Comment by NegativeK** [comment]
Gahh. That makes 100% sense, but Github won't let me add myself as a potential Reviewer.

I can set you as the Reviewer as long as you know that you've got no obligation (you probably already do!) and I might self-review at some point.

**Comment by mrjones-plip** [comment]
cools! and you can make  comments on any PR in the repo - I think it's kinda assumed the author will heavily participate in the PR

File: README.md [hunk 1]
```diff
@@ -58,8 +58,13 @@ python3 camdaemon
 python3 -m flask --app main run -h 0.0.0.0 --reload
 ```
 
-Linting and type checking can be done with:
+If you have make installed, there is a Makefile for development. To do linting
+and type checking, you can run:
 ```
-ruff check *.py
+make static_check
```
Comment (mrjones-plip) [inline README.md]: making this in sync with `Makefile`: 

```suggestion
make static_checking
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+VENV := . venv/bin/activate
+
+types:
+	${VENV} && mypy *.py
+	
+lint:
+	${VENV} && ruff check
+
+static_checking: lint types
```
Comment (mrjones-plip) [inline Makefile]: Optionally simplify this as it feels awkward to repeatedly enable the virtenv when we don't actually need it (made change locally and confirmed this works).  

optional!

```suggestion
types:
        ./venv/bin/mypy *.py

lint:
        ./venv/bin/ruff check

static_checking: lint types
```

**Review by mrjones-plip — APPROVED** [review]
Approving as this works with a few minor tweaks (mainly `make static_check` -> `make static_checking`) - see comments below.

Hot take:  skip `make` entirely.  This is a lot to to just run two lines of bash.  Or maybe use `lint.sh` instead of the `Makefile`?  

No biggie! Doer Decides!

**Comment by mrjones-plip** [comment]
oh yes - I assume this is accurate to not pass linting?  or we can fix in another PR if it's not accurate?

```
root@picam:/home/picam/picamserver# make static_checking
./venv/bin/ruff check
All checks passed!
./venv/bin/mypy *.py
camdaemon.py:6: error: Cannot find implementation or library stub for module named "picamera2"  [import-not-found]
camdaemon.py:6: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
config.py: error: Source file found twice under different module names: "picamserver.config" and "config"
config.py: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#mapping-file-paths-to-modules for more info
config.py: note: Common resolutions include: a) adding `__init__.py` somewhere, b) using `--explicit-package-bases` or adjusting MYPYPATH
Found 2 errors in 2 files (errors prevented further checking)
make: *** [Makefile:3: types] Error 2
```

**Comment by NegativeK** [comment]
mypy errors documented in #28 .

I know it’s make abuse, but it’s kind of habit for me at this point!

**Pull request merged**